aten/src/ATen/native/cpu/layer_norm_kernel.cpp

```diff
@@ -53,12 +50,12 @@ void LayerNormKernelImplInternal(
     for (const auto i : c10::irange(start, end)) {
       const T* X_ptr = X_data + i * N;
       T* Y_ptr = Y_data + i * N;
-      T_ACC mean_val;
-      T_ACC rstd_val;
+      T mean_val;
+      T rstd_val;
```

Comment:
is the loss of ACC here intentional? IT isn't obvious to me that this template will never be instantiated with half precision type.

Reply:
this PR adds a explicit specification of `<BFloat16>` for the base template of `template <typename T>LayerNormKernelImplInternal` because of the need to dispatch to `layer_norm_kernel_mixed_type` when input is `BFloat16` and params is `BFloat16` or `float`.

So when input is `BFloat16`, it won't go to the base template of `LayerNormKernelImplInternal`. Previously `BFloat16` will also go to the base template so we need to use `T_ACC` in it. Now `T_ACC` would not been needed since `T` will only be float or double.